lib/complete-provider.coffee

```diff
@@ -53,11 +58,19 @@ class CompleteProvider extends Provider
     'then', 'where'
   ]
 
-  initialize: (@editorView, @manager) ->
+  constructor: (@editor, @manager) ->
+    if (@editor.haskellCompletionProvider && @editor.haskellCompletionProvider != this)
+      # shouldn't happen, but log just in case
+      console.log "weird: editor already has a haskell completion provider"
+      @editor.haskellCompletionProvider.dispose()
```

Comment:
Don't have a provider per editor :)

See [here](https://github.com/atom-community/autocomplete-plus/blob/074a1a7a49bf806a0e830432ac997c57a2c1d08a/lib/fuzzy-provider.coffee#L16), [here](https://github.com/atom-community/autocomplete-plus/blob/074a1a7a49bf806a0e830432ac997c57a2c1d08a/lib/fuzzy-provider.coffee#L10-L11) and [here](https://github.com/atom-community/autocomplete-plus/blob/074a1a7a49bf806a0e830432ac997c57a2c1d08a/lib/fuzzy-provider.coffee#L22-L42) for how FuzzyProvider does this.